torch/_torch_docs.py

```diff
@@ -1242,14 +1242,14 @@ def merge_dicts(*dicts):
 is ``True`` then the returned tensor will require a gradient, and if :attr:`obj` is
 also a tensor with an autograd history then the returned tensor will have the same history.
 
-When :attr:`obj` is not a tensor, NumPy Array, or DLPack capsule but implements Python's
+When :attr:`obj` is not a tensor, NumPy array, or DLPack capsule but implements Python's
 buffer protocol then the buffer is interpreted as an array of bytes grouped according to
 the size of the datatype passed to the :attr:`dtype` keyword argument. (If no datatype is
 passed then the default floating point datatype is used, instead.) The returned tensor
 will have the specified datatype (or default floating point datatype if none is specified)
 and, by default, be on the CPU device and share memory with the buffer.
 
-When :attr:`obj` is none of the above but a scalar or sequence of scalars then the
+When :attr:`obj` is none of the above but a Python or NumPy scalar, or a sequence of scalars then the
```

Comment:
This is a little confusing. The current documentation defines the follow priorities:

- tensor, NumPy array, DLPack capsule
- object implementing Python's buffer protocol
- scalar or sequence of scalars

Doesn't this PR want to add NumPy scalar to the first category, taking precedence over objects that implement the buffer protocol? What happens if a sequence of NumPy scalars is given? Also, this paragraph says that the datatype of the returned tensor is inferred from the scalar values, but for a NumPy scalar isn't the returned tensor's dtype mapped from the NumPy scalar's dtype?

Reply:
> Doesn't this PR want to add NumPy scalar to the first category, taking precedence over objects that implement the buffer protocol?

Yes, that's right.

> What happens if a sequence of NumPy scalars is given?

They are treated as a Python sequence (i.e. datatype is inferred only if it's not explicitly specified).

> Also, this paragraph says that the datatype of the returned tensor is inferred from the scalar values, but for a NumPy scalar isn't the returned tensor's dtype mapped from the NumPy scalar's dtype?

That's correct.
What if we added a new paragraph before this last one, like: 

_"When obj is a NumPy scalar, the returned tensor will be a 0-dimensional tensor that lives on the CPU device and doesn't share its memory (i.e. `copy=True`). Its datatype won't change unless otherwise specified."_

Reply:
Seems pretty reasonable. Small tweak suggestion:

"When obj is a NumPy scalar, the returned tensor will be a 0-dimensional tensor on the CPU and that doesn't share its memory (i.e. copy=True). By default datatype will be the PyTorch datatype corresponding to the NumPy's scalar's datatype."

?